arch/xtensa/core/xtensa-asm2.c

```diff
@@ -255,6 +255,10 @@ void *xtensa_excint1_c(int *interrupted_stack)
 			reason = bsa[BSA_A2_OFF/4];
 			/* Skip ILL to RETW */
 			bsa[BSA_PC_OFF/4] += 3;
+			/* Prevents xtensa_arch_except from returning in ISR context. */
+			if ((arch_curr_cpu()->nested > 1) && (reason == K_ERR_STACK_CHK_FAIL)) {
+				arch_curr_cpu()->nested = 1U;
```

Comment:
This needs more explaining on why this is here, how this works, and what kind of issues we need to be aware of. Next time someone else looking at this would be confused.